src/App.tsx

```diff
@@ -15,36 +16,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = '',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goodsList, setGoodsList] = useState<string[]>(goodsFromServer);
+  const [sortedBy, setSortedBy] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortAlphabetically = () => {
+    if (isReversed) {
+      setGoodsList([...goodsList].sort().reverse());
+    } else {
+      setGoodsList([...goodsList].sort());
+    }
+
+    setSortedBy(SortType.Alphabetically);
+  };
+
+  const sortByLength = () => {
+    if (isReversed) {
+      setGoodsList([...goodsList].sort((a, b) => b.length - a.length));
+    } else {
+      setGoodsList([...goodsList].sort((a, b) => a.length - b.length));
+    }
+
+    setSortedBy(SortType.Length);
+  };
+
+  const reversed = () => {
+    setGoodsList([...goodsList].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const reseted = () => {
+    setGoodsList(goodsFromServer);
+    setSortedBy(SortType.None);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={sortAlphabetically}
+          className={`button is-info  ${sortedBy === SortType.Alphabetically ? '' : 'is-light'}`}
+        >
```

Comment:
in a more React-ish sense it would be better to have a single component (modifiable by its props) for every button